### PR TITLE
Add a PCAP based testbed

### PR DESCRIPTION
**Pull request by mertyildiran**
Adds a test script (consists of Bash and Python) that downloads a bunch of PCAP files from a Google Cloud bucket, runs the tapper against the PCAP files and generates data file named `suite.json` to lock the state.

Run `make test-run-update` to update `suite.json` and run `make test-run` to validate against that file.

Adds `test-lint` Makefile rule to run linter checks locally with ease.

Improves request-response matching logic of HTTP and Redis.

Fixes some interface conversion errors in AMQP and Kafka that's found through those PCAP files.

Introduces an environment variable named `MIZU_TEST` to alter the certain default behavior of Mizu agent, in the test environment to achieve more consistent behavior.

The testing approach that's introduced by this PR, is planned to be included into the CI in the later PRs.

### CODE REVIEW

File: tests/run.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+PCAPS_DIR=tests/pcaps
+[ "$(ls -A --ignore='.??*' tests/pcaps)" ] && echo "Skipping downloading PCAPs" || gsutil -m cp gs://static.up9.io/mizu/test-pcap/\*.pcap $PCAPS_DIR
```
Comment (IgorGov) [inline tests/run.sh:4]: Getting the following error when running:

`ls: unrecognized option `--ignore=.??*'`

Comment (IgorGov) [reply tests/run.sh:4]: Additional error:
`tests/run.sh: line 19: timeout: command not found
Traceback (most recent call last):
  File "/Users/igorgov/dev/mizu/tests/test.py", line 17, in <module>
    import websocket
ModuleNotFoundError: No module named 'websocket'
tests/run.sh: line 27: kill: (-64291) - No such process
make: *** [test-run] Error 1`

Comment (IgorGov) [reply tests/run.sh:4]: should we have a requirements.json, and part of the script create venv and install the requirements into it?

**Comment by mertyildiran** [comment]
The parts of this PR related to PCAP-based testing approach is bad because there are too much asynchronous code. The changes in HTTP, AMQP, Kafka and Redis are good.

Merge https://github.com/up9inc/mizu/pull/767 first. Then maybe close this PR and create a new one with good changes.

File: Makefile [hunk 2]
```diff
@@ -83,6 +83,23 @@ clean-cli:  ## Clean CLI.
 clean-docker:
 	@(echo "DOCKER cleanup - NOT IMPLEMENTED YET " )
 
+test: test-lint test-run
```
[the remark this answers is not in the document]
Comment (IgorGov) [reply Makefile:86]: Agree.
"Tapper" is a separate component which is used in UP9.app, Mizu community and Mizu Enterprise. 
We should test it as component, including unit tests for specific functionalities (like protocols: kafka https, etc..) and integration tests for all tapper.

**Comment by mertyildiran** [comment]
Closing in favor of other PRs.

**Pull request closed without merging**